.github/workflows/build.yml

```diff
@@ -335,29 +354,49 @@ jobs:
 
       - name: Build iOS app
         run: |
-          ./Scripts/build.sh
+          ./Scripts/build.sh ${{ env.BUILD_VARIANT }} "generic/platform=iOS Simulator"
+
+      - name: Prepare APP files for upload to GitHub
+        if: env.BUILD_VARIANT == 'Simulator'
```

Comment:
Looking at the file as a whole I think there's two things we could do to spare all of these step checks and duplicates:

1. Preliminary step to set vars for file paths, have scripts check environment variables for simulator specific commands. For the purpose of this PR we can focus on Simulator related changes but this can be applied to all Beta / Prod steps too. Most of them could later be moved to GitHub Environments.
2. I was going to suggest rearranging steps and adding an early exit step but, seems like there's [no support yet ](https://github.com/actions/runner/issues/662) for an early exit with a success state. We can only stop with an error `exit 1` or cancel `gh run cancel ${{ github.run_id }} |  gh run watch ${{ github.run_id }}`. 🤔

Reply:
Hm...I'm not sure I understand how abstracting out the file paths or simulator-specific commands would solve the problem 🤔 it's more that just, if we're building for the simulator we don't need to do a bunch of stuff (but we still could, it would just not have an effect), and the `xcodebuild` invocation is slightly different. But I'm also not sure of the full capabilities of GitHub Environments in that regard.